addons/payment/models/account_payment.py

```diff
@@ -25,6 +26,34 @@ class AccountPayment(models.Model):
         compute='_compute_use_electronic_payment_method',
         help='Technical field used to hide or show the payment_token_id if needed.'
     )
+    # == Fields used for traceability ==
+    source_payment_id = fields.Many2one(
+        string="Source Payment",
+        comodel_name='account.payment',
+        help="The source payment of related refund payments",
+        related='payment_transaction_id.source_transaction_id.payment_id',
+        readonly=True,
+        store=True)  # needed for the groupby in the `_compute_refunds_count`
+    refund_payment_ids = fields.One2many(
+        string="Refund Payments",
+        comodel_name='account.payment',
+        inverse_name='source_payment_id',
+        readonly=True)
+    refunds_count = fields.Integer(string="Refunds Count", compute='_compute_refunds_count')
+
+    def _compute_available_amount_for_refund(self):
+        for payment in self:
+            if payment.payment_transaction_id.sudo().acquirer_id.type_refund_supported:
+                refund_txs = payment.env['payment.transaction'].search([
+                    ('source_transaction_id', '=', payment.payment_transaction_id.id),
+                    ('operation', '=', 'refund'),
+                    ('state', 'in', ['draft', 'pending', 'authorized', 'done'])
```

Comment:
@pierre-pde 
The better processing of errors is a must be. 
The problems are:
A) The 5 minutes gap between the demand of a refund, and the reception of a notification. A transaction is in draft.
B) A bad setup for a notification webhook, where the users don't receive success nor cancel notification. In both cases, the transaction will never go further than a draft. But if it's a success, it should be counted in the available amount, if not, it shouldn't. And Odoo has no way to know which one.

For this, there are different possibilities:

1) We don't take the refund in draft mode into the refunded amount. This mean that for case A, someone else can think the refund isn't done and do it again if they don't check the logger. If the total demand of refund is superior to the available amount, there will be a failure notification for the last one. Else, both will be processed. Or, in case B, the users will only see the problem if they pay attention to the log/time/difference with the refund amount available.  But since we won't count the draft transaction for the available amount to refund, the users will be able to ask for a refund that will bypass the draft transaction. This way, if a cancel (for instance for the full amount) isn't notified, the user can still ask again for it. They can ask refunds until the initial amount is really refunded. After this point, they can still ask, but Adyen won't process it anymore.

2) Same as 1, but we add a field with the "pending refund" to show the amount of draft transactions, to have a visual in the wizard. But it's another field for what should be a simple wizard. We could only show it when the amount is > 0. So, the users could see something like : 
    - Initial amount: 10.0
    - Refunded amount: 3
    - Refund amount pending: 2
    - Available amount to refund: 7
But, it may be disturbing for the user.

3) Same as 1 + if there is a draft transaction, make a warning with something like "There is a refund pending for this payment. Wait a moment for it to be processed. If this message is still there, please verify that your Adyen's notifications are correctly set-up."